docker/package/wizard_structure.py

```diff
@@ -385,3 +585,137 @@ def fill_baking_config(self):
+
+        self.config["remote_host"] = rsu.group(1)
+        self.config["remote_key"] = rsu.group(2)
+
+    def get_current_head_level(self):
+        response = urllib.request.urlopen(
+            self.config["node_rpc_addr"] + "/chains/main/blocks/head/header"
+        )
+        return str(json.load(response)["level"])
+
+    # Check if an account with the baker_alias alias already exists, and ask the user
+    # if it can be overwritten.
+    def check_baker_account(self):
+        baker_alias = self.config["baker_alias"]
+        baker_key_value = get_key_address(self.get_tezos_client_options(), baker_alias)
+        if baker_key_value is not None:
+            value, address = baker_key_value
+            print()
+            print("An account with the '" + baker_alias + "' alias already exists.")
+            print("Its current value is", value)
+            print("With a corresponding address:", address)
+
+            return yes_or_no(
+                "Would you like to import a new key and replace this one? <y/N> ", "no"
+            )
+        else:
+            return True
+
+    def import_key(self, key_mode_query):
+
+        baker_alias = self.config["baker_alias"]
+        tezos_client_options = self.get_tezos_client_options()
+
+        valid_choice = False
+        while not valid_choice:
+
+            try:
+                self.query_step(key_mode_query)
+
+                if self.config["key_import_mode"] == "secret-key":
+                    self.query_step(secret_key_query)
+                    proc_call(
+                        f"sudo -u tezos {suppress_warning_text} tezos-client {tezos_client_options} "
+                        f"import secret key {baker_alias} {self.config['secret_key']} --force"
+                    )
+                elif self.config["key_import_mode"] == "remote":
+                    self.fill_remote_signer_infos()
+
+                    tezos_client_options = self.get_tezos_client_options()
+                    proc_call(
+                        f"sudo -u tezos {suppress_warning_text} tezos-client {tezos_client_options} "
+                        f"import secret key {baker_alias} remote:{self.config['remote_key']} --force"
+                    )
+                elif self.config["key_import_mode"] == "json":
+                    self.query_step(json_filepath_query)
+                    json_tmp_path = shutil.copy(self.config["json_filepath"], "/tmp/")
+                    proc_call(
+                        f"sudo -u tezos {suppress_warning_text} tezos-client {tezos_client_options} "
+                        f"activate account {baker_alias} with {json_tmp_path} --force"
+                    )
+                    try:
+                        os.remove(json_tmp_path)
+                    except:
+                        pass
+                else:
+                    print("Please open the Tezos Baking app on your ledger.")
```

Comment:
So... this has no escape. If I get here by mistake (which isn't too unlikely given that this is, correctly, the default) there is no way to abort, except by closing the whole wizard.

Maybe there should be a way to stop the waiting and go back to the previous choice?